core/src/main/java/hudson/model/FileParameterValue.java

```diff
@@ -241,11 +257,120 @@ private File getFileParameterFolderUnderBuild(AbstractBuild<?, ?> build) {
 
     /**
      * Default implementation from {@link File}.
+     *
+     * @deprecated use {@link FileItemImpl2}
      */
-    public static final class FileItemImpl implements FileItem {
-        private final File file;
+    @Deprecated
+    public static final class FileItemImpl implements org.apache.commons.fileupload.FileItem {
```

Comment:
This is basically only consumed by tests, and might even make more sense in the test harness, though I am not tackling that here. However, to retain compatibility at compile time and in tests, it has to retain the old API. I am providing a `FileItemImpl2` that implements the new API for use by core itself and implement this in terms of that (through the delegate pattern) in order to reuse code.